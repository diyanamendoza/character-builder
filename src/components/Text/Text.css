@import url('https://fonts.googleapis.com/css2?family=Alegreya+Sans+SC:ital,wght@1,800&family=Oswald:wght@600&display=swap');

.text {
  display: flex;
  gap: 10px;
  margin-bottom: 30px;
}

button {
  border-style: none;
  border-radius: 10px;
  font-family: 'Alegreya Sans SC', sans-serif;
  background-color: black;
  color: white;
  cursor: pointer;
}

button:active {
  background-color: white;
  color: black;
}
