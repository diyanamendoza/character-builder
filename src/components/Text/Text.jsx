import React from 'react';
import './Text.css';

export default function Text({ text, setText, handleClick }) {
  return (
    <div className="text">
      <label>
        Motto
        <input
          value={text}
          style={{ marginLeft: 10, width: 200 }}
          onChange={(e) => setText(e.target.value)}
        />
      </label>
      <button type="button" onClick={handleClick}>
        Submit
      </button>
    </div>
  );
}
