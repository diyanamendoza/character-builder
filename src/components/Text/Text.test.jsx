import { screen, render } from '@testing-library/react'
import Text from './Text.jsx'

it('should render text input component', () => {
  const { container } = render(<Text text="tasty dumplings" />)
  const mottoInput = screen.getByLabelText(/Motto/)
  const mottoButton = screen.getByRole('button')

  screen.debug()

  expect(mottoInput).toBeInTheDocument()
  expect(mottoButton).toBeInTheDocument()
  expect(container).toMatchSnapshot()
})
