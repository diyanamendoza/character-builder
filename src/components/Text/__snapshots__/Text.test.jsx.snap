// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render text input component 1`] = `
<div>
  <div
    class="text"
  >
    <label>
      Motto
      <input
        style="margin-left: 10px; width: 200px;"
        value="tasty dumplings"
      />
    </label>
    <button
      type="button"
    >
      Submit
    </button>
  </div>
</div>
`;
